Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/ScannerExtensions.kt

```diff
@@ -37,6 +37,12 @@ fun Throwable.toQrCodeErrorDialogBuilder(context: Context): MaterialAlertDialogB
                         context.openUrl(R.string.error_dcc_already_registered_faq_link)
                     }
                 }
+                throwable.isBlocked -> {
+                    setTitle(R.string.error_dcc_in_blocklist_title)
+                    setNegativeButton(R.string.error_dcc_in_blocklist_faq) { _, _ ->
+                        context.openUrl(R.string.dcc_signature_validation_dialog_faq_link)
+                    }
+                }
```

Comment:
Dialog buttons should be reversed in order
<img width="961" alt="Screenshot 2021-11-09 at 15 37 14" src="https://user-images.githubusercontent.com/25054729/140944229-d8986ed4-0902-48ee-a7b0-cb6a9e9c21fa.png">

Look at figma https://www.figma.com/file/KuDEcMSREg3VhZRhC3FXTY/COVID19_Master_2.11?node-id=37357%3A27922

Reply:
AFIK that's iOS dialog, and we should rather stick to our material design guidelines where the button order is as follows: 
<img width="1002" alt="image" src="https://user-images.githubusercontent.com/2919625/140954214-504520fa-9665-4744-bb0d-c8dff50af019.png">

Reply:
OK, btw I think there is an issue here from before , we should use `setNeutralButton` as other dialogs ,but apparently it is overridden by  ` .setNeutralButton(null, null) // Remove details` and this removes any link in the dialogs. can you please fix it? I think the following would fix it 
```kotlin
    return toErrorDialogBuilder(context).apply {
        if (throwable is InvalidHealthCertificateException) {
            when {
                throwable.isCertificateInvalid ->
                    setNeutralButton(R.string.error_button_dcc_faq) { _, _ ->
                        context.openUrl(R.string.error_button_dcc_faq_link)
                    }

                throwable.isSignatureInvalid -> {
                    setTitle(R.string.dcc_signature_validation_dialog_title)
                    setNeutralButton(R.string.dcc_signature_validation_dialog_faq_button) { _, _ ->
                        context.openUrl(R.string.dcc_signature_validation_dialog_faq_link)
                    }
                }
                throwable.isAlreadyRegistered -> {
                    setNeutralButton(R.string.error_button_dcc_faq) { _, _ ->
                        context.openUrl(R.string.error_dcc_already_registered_faq_link)
                    }
                }
                throwable.isBlocked -> {
                    setTitle(R.string.error_dcc_in_blocklist_title)
                    setNeutralButton(R.string.error_dcc_in_blocklist_faq) { _, _ ->
                        context.openUrl(R.string.dcc_signature_validation_dialog_faq_link)
                    }
                }
            }
        } else {
            setNeutralButton(null, null) // Remove details
        }
    }
```

Reply:
other `InvalidHealthCertificateException` shouldn't have an FAQ button according to the Figma design. It has been changed by this PR https://github.com/corona-warn-app/cwa-app-android/pull/4171 by adding `setNeutralButton(null, null)` but I can probably remove `setNeutralButton()` directly from `toErrorDialogBuilder()` method when we don't need it anymore

Reply:
> other InvalidHealthCertificateException shouldn't have an FAQ button according to the Figma design.
Where do you see that ?
#4171  was only to remove the `details` button not FAQ button ,I think this wasn't intended ,right?
/cc @BMItr

Reply:
yes, #4171 was only to remove the details button (and stay on the screen to scan again)

Reply:
updated